0.1.29 doesn't build against musl libc.

For example, building in an alpine container:

```
6.767 error[E0425]: cannot find function `preadv2` in crate `libc`
6.767     --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/jobserver-0.1.29/src/unix.rs:598:15
6.767      |
6.767 598  |           libc::preadv2(
6.767      |                 ^^^^^^^ help: a function with a similar name exists: `preadv`
6.767      |
6.767     ::: /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.153/src/unix/linux_like/linux/mod.rs:4868:13
6.767      |
6.767 4868 | /             pub fn preadv(
6.767 4869 | |                 fd: ::c_int,
6.767 4870 | |                 iov: *const ::iovec,
6.767 4871 | |                 iovcnt: ::c_int,
6.767 4872 | |                 offset: ::off_t,
6.767 4873 | |             ) -> ::ssize_t;
6.767      | |__________________________- similarly named function `preadv` defined here
6.767
6.768 error[E0425]: cannot find value `RWF_NOWAIT` in crate `libc`
6.768     --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/jobserver-0.1.29/src/unix.rs:606:19
6.768      |
6.768 606  |             libc::RWF_NOWAIT,
6.768      |                   ^^^^^^^^^^ help: a constant with a similar name exists: `IPC_NOWAIT`
6.768      |
6.768     ::: /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.153/src/unix/linux_like/linux/mod.rs:2102:1
6.768      |
6.768 2102 | pub const IPC_NOWAIT: ::c_int = 0o4000;
6.768      | ----------------------------- similarly named constant `IPC_NOWAIT` defined here
6.768
6.821 For more information about this error, try `rustc --explain E0425`.
```

https://docs.rs/libc/latest/x86_64-unknown-linux-musl/libc/index.html?search=preadv2

--- a/src/unix.rs
+++ b/src/unix.rs
@@ -287,7 +287,10 @@ impl Client {
 
         // On Linux, we can use preadv2 to do non-blocking read,
         // even if `O_NONBLOCK` is not set.
-        #[cfg(target_os = "linux")]
+        //
+        // TODO: musl libc supports preadv2 since 1.2.5, but `libc` crate
+        // hasn't yet added it.
+        #[cfg(all(target_os = "linux", target_env = "gnu"))]
         {
             let read = self.read().as_raw_fd();
             loop {
@@ -577,7 +580,9 @@ extern "C" fn sigusr1_handler(
     // nothing to do
 }
 
-#[cfg(target_os = "linux")]
+// This should be available for all linux targets,
+// though only [`non_blocking_read`] currently uses it so adding gnu cfg.
+#[cfg(all(target_os = "linux", target_env = "gnu"))]
 fn cvt_ssize(t: libc::ssize_t) -> io::Result<libc::ssize_t> {
     if t == -1 {
         Err(io::Error::last_os_error())
@@ -586,7 +591,7 @@ fn cvt_ssize(t: libc::ssize_t) -> io::Result<libc::ssize_t> {
     }
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(all(target_os = "linux", target_env = "gnu"))]
 fn non_blocking_read(fd: c_int, buf: &mut [u8]) -> io::Result<usize> {
     static IS_NONBLOCKING_READ_UNSUPPORTED: AtomicBool = AtomicBool::new(false);
 

